### PR TITLE
11 completed, added function to check if topic exists, changed curren…

### PR DESCRIPTION
**Pull request by Antonio-Bogalho**
…t logic as previously it would pass no matter what query was, removed pointless tests as from ticket 10 feedback

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -350,20 +350,46 @@ describe("GET /api/users", () => {
...
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(12);
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("article_id", expect.any(Number));
+          expect(article).toHaveProperty("author", expect.any(String));
+          expect(article).toHaveProperty("title", expect.any(String));
+          expect(article).toHaveProperty("topic", "mitch");
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:366]: 👍 ⭐

File: models/articles.model.js [hunk 2]
```diff
@@ -15,29 +15,34 @@ exports.selectArticleid = (article_id) => {
...
+  if (topic) {
+    queryStr += ` WHERE articles.topic = $1`;
+    queryParams.push(topic);
+  }
+
+  queryStr += ` GROUP BY articles.article_id ORDER BY articles.created_at DESC;`;
+
+  return db.query(queryStr, queryParams).then((result) => {
+    return result.rows.map((article) => {
+      return { ...article, comment_count: Number(article.comment_count) };
```
Comment (nc-jo-watson) [inline models/articles.model.js:43]: Good solution! You can also do this in your SQL command, take a look at CAST()

**Review by nc-jo-watson — COMMENTED** [review]
Nice Antonio, just a few suggestions from me!
This is the code for ticket 12, though 👀 Make sure to go back and do ticket 11, which will give you more queries to handle on this endpoint!

**Comment by Antonio-Bogalho** [comment]
Hi Jo! ahahaha I was scrolling through the questions to see what's
ahead,and I thought I was doing 11 the whole time, now when I was moving on
to the next question and I saw 13 I realized, I'm working on it and will
reupload the solutions thank you for the feedback 😆

**Pull request merged**